source/drivers/MicroBitFlash.cpp

```diff
@@ -101,15 +108,19 @@ int MicroBitFlash::need_erase(uint8_t* source, uint8_t* flash_addr, int len)
   * Erase an entire page
   * @param page_address address of first word of page
   */
-void MicroBitFlash::erase_page(uint32_t* pg_addr) 
+uint8_t MicroBitFlash::erase_page(uint32_t* pg_addr) 
 {
     if (ble_running())
     {
         flash_op_complete = false;
         while(1)
         {
-            if (sd_flash_page_erase(((uint32_t)pg_addr)/PAGE_SIZE) == NRF_SUCCESS)
+            uint8_t ret = sd_flash_page_erase(((uint32_t)pg_addr)/PAGE_SIZE);
+            if (ret == NRF_SUCCESS) {
                 break;
+            } else {
+                return ret;
+            }
```

Comment:
@microbit-sam  Looks like some code paths don't explicitly return a value here? Can we either update to be consistent or revert to earlier "void" function prototype if the return value is no longer needed?